src/components/common/WarningMessageWithIcon.test.tsx

```diff
@@ -0,0 +1,28 @@
+import { test, expect, afterEach } from 'vitest';
+import { cleanup, screen } from '@testing-library/react';
+import { Warning } from './WarningMessageWithIcon';
+import { renderWithRouter } from 'libs/testing-library';
+
+afterEach(async () => {
+  cleanup(); // Clear the screen after each test
+  window.history.replaceState(null, 'root', '/');
+});
+
+const component = () => {
+  return (
+    <Warning htmlFor="" message="" className="" isError={true}>
+      Example warning
+    </Warning>
+  );
+};
+
+test('Test warning component', async () => {
+  // ARRANGE
+  await renderWithRouter({ component });
```

Comment:
Feels like this is the kind of component that do not need the router. I think it's nice if we can have an example of a page (need the router) and a reusable component (don't need the router).

Reply:
You're right, this was just while I was testing out the router, We already have MarketPriceIndication as an example without the router, so I'll remove this file